### PR TITLE
improvement(zone): refactor `tags` attribute to `Array(String)`

### PR DESCRIPTION
**Pull request by caspiano**

### CODE REVIEW

File: src/models/zone.cr [hunk 1]
```diff
@@ -14,7 +14,7 @@ module PlaceOS::Model
 
     attribute name : String, es_type: "keyword"
     attribute description : String
-    attribute tags : String
+    attribute tags : Array(String) = ->{ [] of String }
```
Comment (KimBurgess) [inline src/models/zone.cr:17]: Rather than being an array, can it be a expressed as a `Set(String)` (if supported)?

Comment (caspiano) [reply src/models/zone.cr:17]: depends on whether the rethinkdb driver has implemented `Set` serialisation (which i doubt)

Comment (caspiano) [reply src/models/zone.cr:17]: Doesn't look like it.
I definitely agree with attempting to encode properties into the DB types as much as possible.
Shouldn't be too hard if you want to open a PR on [crystal-rethinkdb](https://github.com/kingsleyh/crystal-rethinkdb).

That driver could use a bit of loving.

Comment (stakach) [reply src/models/zone.cr:17]: We can ensure uniqueness at the ORM level

Comment (caspiano) [reply src/models/zone.cr:17]: I think having a type-level restriction would be nicer

```crystal
a = %w(hello hello hello bye good).to_json # => ["hello","hello","hello","bye","good"]
Set(String).from_json(a).to_json # => ["hello","bye","good"]
```

Comment (caspiano) [reply src/models/zone.cr]: tried it now! looks like it works! @KimBurgess

**Pull request merged**